src/Microsoft.AspNet.WebUtilities/QueryHelpers.cs

```diff
@@ -20,8 +20,7 @@ public static class QueryHelpers
         /// <returns>The combined result.</returns>
         public static string AddQueryString([NotNull] string uri, [NotNull] string name, [NotNull] string value)
         {
-            bool hasQuery = uri.IndexOf('?') != -1;
-            return uri + (hasQuery ? "&" : "?") + UrlEncoder.Default.UrlEncode(name) + "=" + UrlEncoder.Default.UrlEncode(value);
+            return AddQueryString(uri, new Dictionary<string, string> { { name, value } });
```

Comment:
They're not cheap, but we also don't want to duplicate a lot of code... I'm not sure that this method is much of a hot path.

The only other thought I have that doesn't duplicate code is to create a new degenerate dictionary type called `SingleValueDictionary` that takes in `string key, string value` and basically hard-codes all the `IDictionary` stuff to just be aware of the single key/value pair. Probably not a lot of code to write, but not sure it's worth it.